ui/perfherder/compare/ComparePageLabel.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Button, Input, InputGroup } from 'reactstrap';
+
+export default class ComparePageLabel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      inEditMode: false,
+      pageLabel: props.pageLabelQueryParam || props.label,
+      newPageLabel: props.pageLabelQueryParam || props.label,
+    };
+  }
+
+  goToEditMode = () => {
+    this.setState({
+      inEditMode: true,
+    });
+  };
+
+  keydownListener = async event => {
```

Comment:
I think you should also update the title when someone removes focus (clicks elsewhere on the page). And if the new value didn't change you don't need to call `updateParams`. 

Also, it might be clearer to other users what this feature is for if you have a clipboard or similar icon in the upper right corner of the title on hover in addition to the pointer cursor and title tooltip (like with the permalinks) so people don't click on it quickly thinking it's a link.

Reply:
Done. The work was patched into a new commit.